windows/test-app.ts

```diff
@@ -0,0 +1,300 @@
+const testAppNodeModulePath = findClosestPathTo(path.join('node_modules', 'react-native-test-app'));
+const srcRootPath = path.join(testAppNodeModulePath, windowsDir);
+const destPath = path.resolve('');
+
+
+function findClosestPathTo(fileOrDirName: string): string {
+    var basePath = path.resolve('');
+    var rootDirectory = basePath.split(path.sep)[0] + path.sep;
+
+    while (basePath !== rootDirectory) {
+        const candidatePath = path.join(basePath, fileOrDirName)
+        if (fs.existsSync(candidatePath)) {
+            return path.relative('', candidatePath);
+        }
+        //Get parent folder
+        basePath = path.dirname(basePath)
+    }
+
+    return null
+}
+
+/**
+ * Text to replace, + config options
+ */
+export type Replacements = {
+    regExpPatternsToRemove?: RegExp[];
+    [key: string]: any;
+};
+
+function walk(current: string): string[] {
+    if (!fs.lstatSync(current).isDirectory()) {
+        return [current];
+    }
+
+    const files = fs
+        .readdirSync(current)
+        .map(child => walk(path.join(current, child)));
+    const result: string[] = [];
+    return result.concat.apply([current], files);
+}
+
+/**
+ * Get a source file and replace parts of its contents.
+ * @param srcPath Path to the source file.
+ * @param replacements e.g. {'TextToBeReplaced': 'Replacement'}
+ * @return The contents of the file with the replacements applied.
+ */
+export function resolveContents(
+    srcPath: string,
+    replacements: Replacements,
+): string {
+    let content = fs.readFileSync(srcPath, 'utf8');
+
+    Object.keys(replacements).forEach(regex => {
+        content = content.replace(new RegExp(regex, 'g'), replacements[regex]);
+    });
+
+    return content;
+}
+
+// Binary files, don't process these (avoid decoding as utf8)
+const binaryExtensions = ['.png', '.jar', '.keystore'];
+
+/**
+ * Copy a file to given destination, replacing parts of its contents.
+ * @param srcPath Path to a file to be copied.
+ * @param destPath Destination path.
+ * @param replacements: e.g. {'TextToBeReplaced': 'Replacement'}
+ */
+async function copyAndReplace(
+    srcPath: string,
+    destPath: string,
+    relativeDestPath: string,
+    replacements: Replacements = {}
+) {
+    const fullDestPath = path.join(destPath, relativeDestPath);
+    if (fs.lstatSync(srcPath).isDirectory()) {
+        if (!fs.existsSync(fullDestPath)) {
+            fs.mkdirSync(fullDestPath);
+        }
+        return;
+    }
+
+    const extension = path.extname(srcPath);
+    if (binaryExtensions.indexOf(extension) !== -1) {
+        // Binary file
+        copyBinaryFile(srcPath, fullDestPath, err => {
+            if (err) {
+                throw err;
+            }
+        });
+    } else {
+        // Text file
+        const srcPermissions = fs.statSync(srcPath).mode;
+        let content = resolveContents(srcPath, replacements);
+
+        fs.writeFileSync(fullDestPath, content, {
```

Comment:
Do we really need to wait for this write to finish?
```suggestion
        fs.writeFile(fullDestPath, content, {
```